### PR TITLE
Cli & Query for governance xmod

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: gears/src/rest/pagination.rs [hunk 1]
```diff
@@ -25,6 +25,23 @@ pub fn parse_pagination(pagination: Pagination) -> (u32, u8) {
...
+        core_types::query::request::PageRequest {
+            key: _,
+            offset,
+            limit,
+            count_total: _,
+            reverse: _,
+        }: core_types::query::request::PageRequest,
+    ) -> Self {
+        Self {
+            offset: Some(offset as u32),
```
Comment (joneskm) [inline gears/src/rest/pagination.rs:39]: These casts can wrap around, which is probably not the desired behaviour.

File: gears/src/ext.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/// - `T`: The input `Option`'s value type
+/// - `U`: The outputs `Option`'s value type
+/// - `E`: The possible error during the mapping
+pub trait FallibleMapExt<T, U, E> {
+    /// Try to apply a fallible map function to the option
+    fn try_map<F>(self, f: F) -> Result<Option<U>, E>
+    where
+        F: FnOnce(T) -> Result<U, E>;
+}
+
```
Comment (joneskm) [inline gears/src/ext.rs:22]: Can you achieve something similar using collect?

Comment (NYBACHOK) [reply gears/src/ext.rs:22]: It's possible.  Collect `Vec<Result<T, ()>>` iterate over elements and push them to new collection

File: gears/src/rest/pagination.rs (hunk 1 above)
Comment (NYBACHOK) [reply gears/src/rest/pagination.rs:39]: I'll try to change it there

Comment (NYBACHOK) [reply gears/src/rest/pagination.rs]: Okay. I tried to make it `u64`, but we depend on tendermint in [this](https://github.com/NYBACHOK/gears/blob/c97a2a209acf0b192859781d2c6deaa767544451/gears/src/rest/handlers.rs#L69) code, so I have chosen to set max values if couldn't narrow number

File: x/gov/src/params.rs [hunk 3]
```diff
@@ -21,14 +26,17 @@ const DEFAULT_PERIOD: Duration = Duration::from_secs(172800); // 2 days
...
+    pub min_deposit: SendCoins,
     pub max_deposit_period: Duration, // ?
 }
 
 impl Default for DepositParams {
     fn default() -> Self {
         Self {
-            min_deposit: vec![Coin::from_str("10000000stake").expect("default is valid")],
+            min_deposit: SendCoins::new(vec![
+                Coin::from_str("10000000stake").expect("default is valid")
```
Comment (dmvict) [inline x/gov/src/params.rs:37]: Is it from sdk? As I remember we use `uatom`

Comment (NYBACHOK) [reply x/gov/src/params.rs:37]: [It's](https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/x/gov/types/params.go#L51-L55). Think I should change to `uatom`?

**Review by dmvict — APPROVED** [review]

File: x/gov/src/params.rs (hunk 3 above)
Comment (dmvict) [reply x/gov/src/params.rs:37]: I don't know. Maybe we can use this denom.

Comment (joneskm) [reply x/gov/src/params.rs:37]: For now just make it "uatom". We might need a better solution in future.

Comment (NYBACHOK) [reply x/gov/src/params.rs]: Done

**Pull request merged**